src/legacy/ui/public/saved_objects/show_saved_object_save_modal.tsx

```diff
@@ -21,7 +21,12 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import { I18nContext } from 'ui/i18n';
 
-export function showSaveModal(saveModal) {
+interface MinimalSaveModalProps {
+  onSave: (...args: any[]) => Promise<{ id?: string } | { error: Error }>;
```

Comment:
Can you add a comment that explains why the Promise's value would possibly be an object without `id`?